tests/resources/valid/function/definition.mamba

```diff
@@ -3,6 +3,19 @@ def fun_a() =>
     if a and b then print c
     if c or d then print e
 
+class Something
+    def +(self, other: Int): Int => a + b
```

Comment:
As these are inside Class something and override its methods, their return types should actually be `Something`, which can be verified by the type checker.